core/backend/src/ECDb.ts

```diff
@@ -57,7 +57,22 @@ export class ECDb implements IDisposable {
     this._nativeDb.dispose();
     this._nativeDb = undefined;
   }
-
+  /**
+   * Attach a ECDb file to this connection and load and register its schemas.
+   * @param fileName ECDb file name
+   * @param alias identifier for the attached file. This identifer is used a tablespace executing ECSQL queries.
+   */
+  public attachDb(fileName: string, alias: string): void {
```

Comment:
should these be public right out the bat? what is the driver for this?

Reply:
These are exposing the api we already had on native side. I thought I attach the the issue but here it is https://github.com/iTwin/itwinjs-core/issues/7532 Its request by  Clash Studio App where they want to attach db containing clashes to imodel for following. I am still waiting for like to PBI.

1. Query all clashes on an element - along the lines of what you mentioned in the description there.
   - And permutations - all clashes that were suppressed, resolved, assigned to somebody, etc.  
3. Query all clashes on a set of elements returned by a query - very similar to what you have there. 
4. Query all clashes that were resolved/introduced in a named version - join with the clash changesummary view also.

Reply:
Also note some where similar requirement we have for attaching analytical dataset. Where we can combine data from analytical engine with the imodel for which it is calculated. Though bis working group is still working on how schema for that will look like. So these Api will be needed any way as not all data that we compute through external engines will be stored in IModel as it could be sometime 100s of GBs. It also transient which mean as iModel changes that data is recomputed for various scenarios. But we do want to combine that data with IModel for presentation and reporting.

Reply:
Thanks for the clarification. If you can add the above details to the issue that would be great.